### PR TITLE
Charlesmchen/calls stuck on connecting

### PR DESCRIPTION
**Pull request by charlesmchen**
PTAL @michaelkirk

### CODE REVIEW

File: Signal/src/call/CallService.swift [hunk 1]
```diff
@@ -494,19 +505,15 @@ protocol CallServiceObserver: class {
             newCall.callRecord = callRecord
             callRecord.save()
 
-            return
-        }
-
-        guard self.call == nil else {
-            // TODO on iOS10+ we can use CallKit to swap calls rather than just returning busy immediately.
-            Logger.verbose("\(TAG) receivedCallOffer for thread: \(thread) but we're already in call: \(call!)")
+            terminateCall()
```
Comment (michaelkirk) [inline Signal/src/call/CallService.swift]: This is a good catch.

File: Signal/src/call/CallService.swift [hunk 2]
```diff
@@ -470,9 +470,20 @@ protocol CallServiceObserver: class {
 
         let newCall = SignalCall.incomingCall(localId: UUID(), remotePhoneNumber: thread.contactIdentifier(), signalingId: callId)
 
+        guard self.call == nil else {
```
Comment (michaelkirk) [inline Signal/src/call/CallService.swift]: Why this reordering? I think we should prefer *not* to send anything (even a busy signal) in this scenario.

Comment (charlesmchen) [inline Signal/src/call/CallService.swift]: Reverted in https://github.com/WhisperSystems/Signal-iOS/pull/2220

**Comment by charlesmchen** [comment]
The core goal here is to improve logging around the "call stuck in connecting" issue.

**Pull request merged**